src/components/line-chart-audification/line-chart-audification.component.ts

```diff
@@ -35,18 +36,34 @@ export class LineChartAudificationComponent implements OnDestroy, OnChanges {
     return tA11y(AUDIFICATION.INSTRUCTIONS);
   }
 
-  ngOnDestroy() {
-    this.melody?.dispose();
+  get data() {
+    return this.host.data;
   }
 
-  ngOnChanges(changes: SimpleChanges): void {
-    if ('data' in changes) {
-      const values = this.data.map(datum => datum.value);
-      this.domain = this.data.map(d => d.date).sort((a, b) => a.getTime() - b.getTime());
-      this.range = this.data.map(d => d.value).sort();
-      this.melody?.dispose();
-      this.melody = new Melody(values, this.frequencyRange, this.duration, this.handleSeek);
-    }
+  get measureName() {
+    return this.host.measureName;
+  }
+
+  set activeDatum(activeDatum) {
+    this.host.activeDatum = activeDatum;
+  }
+
+  ngOnInit() {
+    this.host.data$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(data => {
+        const values = data.map(datum => datum.value);
+        this.domain = data.map(d => d.date).sort((a, b) => a.getTime() - b.getTime());
```

Comment:
nit: can this sort function be pulled out into a pure util function?

Reply:
Sounds good. I also just realized that the line below that
```ts
this.range = this.data.map(d => d.value).sort();
```
has been wrongly sorting in alphabetical order instead of in numerical order.